doc/api/async_context.md

```diff
@@ -224,13 +224,14 @@ to `asyncLocalStorage.getStore()` will return `undefined` until
 When calling `asyncLocalStorage.disable()`, all current contexts linked to the
 instance will be exited.
 
-Calling `asyncLocalStorage.disable()` is required before the
-`asyncLocalStorage` can be garbage collected. This does not apply to stores
-provided by the `asyncLocalStorage`, as those objects are garbage collected
-along with the corresponding async resources.
+There is no need to call this method in order to get an `asyncLocalStorage`
+instance from being garbage-collected.
 
-Use this method when the `asyncLocalStorage` is not in use anymore
-in the current process.
+When running Node.js with CLI flag [`--no-async-context-frame`][], calling
+`asyncLocalStorage.disable()` is required before the `asyncLocalStorage` itself
+can be garbage collected. However, this does not apply to stores provided by
+the `asyncLocalStorage`, as those objects are garbage collected along with the
+corresponding async resources.
```

Comment:
This second sentence here is rather confusing if you're not familiar with ALS as a whole. I'm not sure it adds a lot of value here.

Reply:
It is an important trait of `disable` that it does not make all entered stores GC-able. It is an existing document sentence that re-organized in this PR:

https://github.com/nodejs/node/blob/fc054bbbb1005c924799c80b07eb0e7ed0278b99/doc/api/async_context.md#L228-L230